File: src/026. Reciprocal cycles/026.lua
function fracCycle(n)
  local r = 1
  local fract = {}
  while r ~= 0 do
    while r < n do
      r = r * 10
    end
    r = r % n
    if r == 0 then
      return 0
    elseif fract[r] == true then
      local len = 0
      for i = 1, 1000 do
        if fract[i] == true then
          len = len + 1
        end
      end
      if n >= 100 then
        len = len * 3
      elseif n >= 10 then
        len = len * 2
      end
      return len
    else
      fract[r] = true
    end
  end
end

local max = 0
local maxNum = 0
for i = 1, 1000 do
  local c = fracCycle(i)
  if c > max then
    max = c
    maxNum = i
  end
end

print(maxNum)

File: src/026. Reciprocal cycles/026.rb
class Fixnum
  def fracCycle
    num = self
    num /= 2 while num % 2 == 0
    num /= 5 while num % 5 == 0
    return 0 if num == 1
	
    cycle = 1
    cycle += 1 while ('9' * cycle).to_i % num != 0
    cycle
  end
end

max = 1
max_cycle = 0
1.upto(1000) do |n|
  if n.fracCycle > max_cycle
    max, max_cycle = n, n.fracCycle
  end
end

puts max
